### PR TITLE
SSCS-10586 reworked court venue lookup

### PR DESCRIPTION
**Pull request by wmarshid**
### JIRA link (if applicable) ###

https://tools.hmcts.net/jira/browse/SSCS-10586

### Change description ###

This change means only one call will be made to reference data during a migration tool run.

- refactored reference data service logic and added/updated tests

**Does this PR introduce a breaking change?**

```
[ ] Yes
[ x ] No
```

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/reform/sscs/migration/service/LocationRefDataService.java [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+
+    private static final String SSCS_COURT_TYPE_ID = "31";
+
+    private final IdamService idamService;
+    private final RefDataApi refDataApi;
+
+    public List<CourtVenue> retrieveCourtVenues() {
+        IdamTokens idamTokens = idamService.getIdamTokens();
+
+        if (idamTokens == null
```
Comment (wmarshid) [inline src/main/java/uk/gov/hmcts/reform/sscs/migration/service/LocationRefDataService.java:26]: @nickwest1 I think you'll like this bit :)

Comment (nickwest1) [reply src/main/java/uk/gov/hmcts/reform/sscs/migration/service/LocationRefDataService.java:28]: hehe, yep, you're right there Waqas :)

File: src/test/java/uk/gov/hmcts/reform/sscs/migration/service/LocationRefDataServiceTest.java [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+        verifyNoInteractions(refDataApi);
+        assertThat(courtVenues).isEmpty();
+    }
+
+    @Test
+    void shouldReturnEmptyList_givenIdamTokensIsEmpty() {
+        when(idamService.getIdamTokens()).thenReturn(IdamTokens.builder()
+            .idamOauth2Token(StringUtils.EMPTY)
+            .serviceAuthorization(StringUtils.EMPTY)
+            .build());
```
Comment (nickwest1) [inline src/test/java/uk/gov/hmcts/reform/sscs/migration/service/LocationRefDataServiceTest.java:55]: Very minor: and very much open to opinion, so just sharing another view:
https://stackoverflow.com/questions/33546124/mockito-given-versus-when

**Review by nickwest1 — APPROVED** [review]

**Review by johnoneill46 — APPROVED** [review]

File: src/test/java/uk/gov/hmcts/reform/sscs/migration/service/LocationRefDataServiceTest.java (hunk 2 above)
Comment (wmarshid) [reply src/test/java/uk/gov/hmcts/reform/sscs/migration/service/LocationRefDataServiceTest.java:55]: I see where you're going with this but I'm a purist at heart.. we should really have a team consensus on this though, I've seen both being used.

**Pull request merged**